pkg/connector/workspace.go

```diff
@@ -74,13 +76,44 @@ func workspaceResource(ctx context.Context, workspace asana.Workspace) (*v2.Reso
 	return ret, nil
 }
 
+func (o *workspaceResourceType) loadAllowedWorkspaces(ctx context.Context) error {
+	o.allowedWorkspacesMutex.Lock()
+	defer o.allowedWorkspacesMutex.Unlock()
+
+	if o.allowedWorkspaces != nil {
+		return nil
+	}
+
+	l := ctxzap.Extract(ctx)
+	l.Info("baton-asana: loading allowed workspaces")
+
+	o.allowedWorkspaces = []string{}
+	workspaceMemberships, err := o.client.AuthCheck(ctx)
+	if err != nil {
+		return err
+	}
+
+	for _, workspaceMembership := range workspaceMemberships {
+		if !workspaceMembership.IsGuest {
+			o.allowedWorkspaces = append(o.allowedWorkspaces, workspaceMembership.Workspace.Gid)
+		}
+	}
+
+	l.Info("baton-asana: loaded allowed workspaces", zap.Any("allowedWorkspaces", o.allowedWorkspaces))
```

Comment:
Should this be debug?

Reply:
Actually, I was planning to remove it after testing in my dev environment. But it may actually be useful if I leave it in and change it to debug.